apps/full-site-editing/package.json

```
@@ -16,49 +16,49 @@
-		"blog-posts-block": "check-npm-client && npm-run-all --parallel blog-posts-block-*",
-		"dev:blog-posts-block": "check-npm-client && npm run blog-posts-block",
-		"build:blog-posts-block": "check-npm-client && NODE_ENV=production npm run blog-posts-block",
-		"starter-page-templates": "check-npm-client && calypso-build --source='starter-page-templates'",
-		"dev:starter-page-templates": "check-npm-client && npm run starter-page-templates",
-		"build:starter-page-templates": "check-npm-client && NODE_ENV=production npm run starter-page-templates",
-		"event-countdown-block": "check-npm-client && calypso-build --source='event-countdown-block'",
-		"dev:event-countdown-block": "check-npm-client && npm run event-countdown-block",
-		"build:event-countdown-block": "check-npm-client && NODE_ENV=production npm run event-countdown-block",
-		"global-styles": "check-npm-client && calypso-build --source='global-styles'",
-		"build:global-styles": "check-npm-client && NODE_ENV=production npm run global-styles",
-		"dev:global-styles": "check-npm-client && npm run global-styles",
-		"jetpack-timeline": "check-npm-client && calypso-build --source='jetpack-timeline'",
-		"dev:jetpack-timeline": "check-npm-client && npm run jetpack-timeline",
-		"build:jetpack-timeline": "check-npm-client && NODE_ENV=production npm run jetpack-timeline",
-		"wpcom-block-editor-nux": "check-npm-client && calypso-build --source='wpcom-block-editor-nux'",
-		"dev:wpcom-block-editor-nux": "check-npm-client && npm run wpcom-block-editor-nux",
-		"build:wpcom-block-editor-nux": "check-npm-client && NODE_ENV=production npm run wpcom-block-editor-nux",
-		"dev": "check-npm-client && node bin/npm-run-build.js --dev",
-		"build": "check-npm-client && node bin/npm-run-build.js --build",
-		"test:js": "check-npm-client && npx wp-scripts test-unit-js --config='bin/js-unit-config.js' --colors",
-		"test:js:help": "check-npm-client && npx wp-scripts test-unit-js --config='bin/js-unit-config.js' --help --colors",
-		"test:js:watch": "check-npm-client && npx wp-scripts test-unit-js --config='bin/js-unit-config.js' --watch --colors",
-		"test:js:update-snapshots": "check-npm-client && npx wp-scripts test-unit-js -u --config='bin/js-unit-config.js' --colors",
-		"test:e2e": "check-npm-client && npx wp-scripts test-e2e --wordpress-base-url='http://localhost:4013' --colors",
-		"test:e2e:watch": "check-npm-client && npm run test-e2e -- --watch",
-		"clean": "check-npm-client && npx rimraf dist full-site-editing-plugin/*/dist",
-		"prebuild": "check-npm-client && npm run clean",
-		"predev": "check-npm-client && npm run clean",
-		"sync:blog-posts-block": "check-npm-client && ./bin/sync-blog-posts-block.sh",
-		"wpcom-sync": "check-npm-client && ./bin/wpcom-watch-and-sync.sh",
-		"prepare": "check-npm-client && npm run sync:blog-posts-block --nodemodules"
+		"full-site-editing": "node ../../bin/check-package-manager.js && calypso-build --source='dotcom-fse'",
```

There is probably a reason you didn't do this, but I'll suggest just in case.

Maybe introduce a `script called check-client` then do 
```suggestion
		"check-client": "node ../../bin/check-package-manager.js",
		"full-site-editing": "yarn check-client && calypso-build --source='dotcom-fse'",
```

---

I consider doing it tbh. It should be `"full-site-editing": "$npm_execpath check-client && calypso-build --source='dotcom-fse'"` (so it uses the same executable that the dev used to run `full-site-editing`).

It feels a bit more obscure ("what is that $npm_execpath magic??"), and I'm not sure it provides a huge benefit over a plain call to node. But I don't have any preference.

What do you think?

---

Well, `$npm_execpath` would fail on Windows. And after a second thought, `yarn/npm check-client` defy the purpose. I think your solution is best.